pkg/storage/azureblob/azureblob.go

```diff
@@ -0,0 +1,155 @@
+package azureblob
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/gomods/athens/pkg/config"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/observ"
+)
+
+type client interface {
+	UploadWithContext(ctx context.Context, path, contentType string, content io.Reader) error
+	BlobExists(ctx context.Context, path string) (bool, error)
+	ReadBlob(ctx context.Context, path string) (io.ReadCloser, error)
+	ListBlobs(ctx context.Context, prefix string) ([]string, error)
+	DeleteBlob(ctx context.Context, path string) error
+	GetTimeout() time.Duration
+}
+
+type azureBlobStoreClient struct {
+	containerURL *azblob.ContainerURL
+	timeout      time.Duration
+}
+
+func newBlobStoreClient(accountURL *url.URL, accountName, accountKey, containerName string, timeout time.Duration) (*azureBlobStoreClient, error) {
+	const op errors.Op = "azure.newBlobStoreClient"
```

Comment:
s/azure/azureblob

Reply:
same for every op here

Reply:
I'm sorry, didn't get you 🤔

Reply:
Should I change the op to `storage.azure.azureblob`

Reply:
sorry my bad
it shoud be `const op errors.Op = "azureblob.newBlobStoreClient"`
same for every errors.Op constant in this file. other files are fine.
the shortcut `s/azure/azureblob` used in the comment means
`s`ubstitue `azure` with `azureblob`

Reply:
@scriptonist @michalpristas mean to change it from `azure.newBlobStoreClient` to `azureblob.newBlobStoreClient` and for other change `azure.*` => `azureblob.*`.

Reply:
Oh Got it now, wasn't accustomed to seeing vim syntax in comments! , hence the confusion.

Reply:
no worries ;-)